### ISSUE TITLE
Configuration to allow more domains in firebase auth emulator?

### ISSUE THREAD

**Issue by jasonford** [firebase/firebase-tools#3104]
https://github.com/firebase/firebase-tools/blob/76726d760ce986c64c5fba30737afea9ea458057/src/emulator/auth/operations.ts#L594

+ 1 for the answer to this TODO question to be yes

Would be super helpful for developing with the emulator on remote machines

**Comment by yuchenshi** [comment]
This comment is a bit misleading here, but SDKs should allow connecting to the Auth Emulator on all domains already. If you notice a difference, please file a bug.

**Comment by jasonford** [comment]
Hey, thanks for the quick reply! On the front end I'm calling

```firebase.auth().useEmulator(AUTH_EMULATOR_HOST)```

and the response when I try to sign in (using the google provider) is a 200, but then the client sdk throws an error with code "auth/unauthorized-domain"

the login request made is: ```https://AUTH_EMULATOR_HOST/identitytoolkit/v3/relyingparty/getProjectConfig?key=KEY```

and the response body is ``{projectId: "12345", authorizedDomains: ["localhost"]}```

That led me to find that line I commented on.

I monkey patched my code by relacing that line with my own "AUTH_EMULATOR_HOST" and emulator login works fine now.

Is there some configuration setting I missed on the front or back end that would allow me authorize another auth emulator host? I was searching for that through the code since I couldn't find it in any documentation, when I came upon that TODO and figured it was time to open an issue.

Also, I was on 8.x of the firebase-tools package on front and back end, so I updated to 9.3.0, but experienced the same behavior.

Thanks for your help!

**Comment by yuchenshi** [comment]
Hi! Thanks for bringing this up. The team is currently working on rewriting the full auth SDK. This issue should be addressed when it launches. In the meantime, your temporary patch for the Auth Emulator is a possible workaround.

To give a bit of context, the Auth Emulator should work on all domains but we cannot make `authorizedDomain` an infinite list of all possible domains, and thus the fix has to be on the SDK side. As you've observed, the Auth Emulator doesn't know about your own host and you had to add it manually. The full rewrite in progress will handle this correctly and will not require patching the Auth Emulator.

**Comment by jasonford** [comment]
Hello, I just want to follow up to see if there is an estimate/planned date for the Auth Emulator rewrite landing?

We still have a monkey patch in our startup script to hot fix the issue for v9.14.0 (posting in case it saves anyone some time):

```sh
  # have to monkey patch firebase-tools for now since the auth emulator is hard-coded to only allow localhost as an "authorizedDomain"
  # TODO: Don't monkey patch if we don't have too (must wait on upstream fix: https://github.com/firebase/firebase-tools/issues/3104)
  # this line just replaces line 445 of serve/node_modules/firebase-tools/lib/emulator/auth/operations.js with our gitpod's host
  sed -i '445s/.*/            "'$(gp url 8080 | cut -c 9-)'"/' node_modules/firebase-tools/lib/emulator/auth/operations.js
```

replace the command "gp url 8080 | cut -c 9-" with whatever domain you want to enable.

**Comment by yuchenshi** [comment]
The new SDK version is in Beta right now: https://firebase.google.com/docs/web/modular-upgrade

Either v9 compact or v9 modular should handle this correctly. If not, please kindly reopen the issue with a minimal repro and we'll look into this.

**Comment by yuchenshi** [comment]
I do believe we've got this fixed in the Firebase JS SDK v9 (modular and compat). The expected behavior is that the Auth Emulator should allow ALL domains, without any need of configuration or patching. If you experience something different, please kindly open a new issue with the details (including SDK versions and actual behaviors) and we can look into this.

P.S. The comments in the Auth Emulator source code has been updated for clarification.
